src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexWriter.cs

```diff
@@ -145,9 +146,14 @@ public RegexCode RegexCodeFromRegexTree(RegexTree tree)
                 CultureInfo culture = (tree.Options & RegexOptions.CultureInvariant) != 0 ? CultureInfo.InvariantCulture : CultureInfo.CurrentCulture;
                 boyerMoorePrefix = new RegexBoyerMoore(leadingSubstring, leadingSubstringCI, rtl, culture);
             }
-            else
+
+            // If we didn't find a single leading substring, or if we found one but we won't be able to use it for a Boyer-Moore
+            // search, try to compute the characters set that might begin the string.
+            if (boyerMoorePrefix is null ||
```

Comment:
```c#
            // If we didn't find a single leading substring, or if we found one but we won't be able to use it for a Boyer-Moore
            // search, try to compute the characters set that might begin the string.
              if (boyerMoorePrefix.NegativeUnicode != null && compiled)) // compilation won't use Boyer-Moore if it has a negative Unicode table
            {
                boyerMoorePrefix = null;
            }
            
            if (boyerMoorPrefix == null)
            {
                       // First we employ a less aggre ..                        
```
?

Reply:
It would actually need to be:
```C#
if (boyerMoorePrefix != null && boyerMoorePrefix.NegativeUnicode != null && compiled)
```
just to avoid the `boyerMoorePrefix = null` in some cases?  Doesn't seem worth it to me.